drivers/adc/adc_common.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/adc.h>
+
+int adc_gain_invert(enum adc_gain gain,
```

Comment:
have you considered using lookup table?

i tried it (https://gist.github.com/nordic-krch/5d4378ee831908705b077e70e770bb08) and gives 84 bytes less ROM (72 vs 156) .

Reply:
Also, if LUT is used then static assert could be used to detect if it is up to date with `enum adc_gain` (enum can be extended with `ADC_GAIN_COUNT` as last item). If that is done then function could return value since error is unlikely (as long as user is using `enum adc_gain` properly). When this function returns value so can `adc_convert_millivolt`.

Reply:
The LUT is fragile since it depends on the order of the values in the enumeration remaining unchanged.  It could be used if there were a test for it.

As I noted in another comment, an assumption that the gain enumeration is valid is not robust since the gain may be read from an uninitialized or corrupted configuration structure.

Reply:
I found a way to resolve my concerns about synchronizing the array index with the behavior: [C99 desginated array initialization](https://en.cppreference.com/w/c/language/array_initialization).

I've updated to use this for a LUT, retaining the validation of the parameter.  76 bytes vs 156, so size is decreased while safety retained.